translator.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * A super simple language translation tool
+ */
+
+// const translationDataExample = {
+//   'hello' : {
+//     'eng': 'Hello!',
+//     'spa': 'Hola!',
+//     'kar': '',
+//     'som': 'Salaan!',
+//     'hmn': 'Nyob zoo!'
+//   }
+// }
+
+class Translator {
+  attr = 'data-translation-id'
+
+  constructor(translations) {
+    this.translations = translations
+  }
+
+  // this converts a `gsx$keyname` to `keyname
+  convertGoogleKey(key) {
+    return 
+  }
+
+  /**
+   * Replace the translatable elements on the page with translated terms (if available)
+   * 
+   * @param {string} lang key for the desired language (example: 'spa')
+   */
+  translate(lang) {
+    this.els = document.querySelectorAll(`[${this.attr}]`) 
+    
+    let current
+    for (let i = 0; i < this.els.length; i++) {
```

Comment:
Is there any reason this can't be done in the functional `els.forEach(` style?

Reply:
`querySelectorAll` returns a `NodeList` (not an array) which doesn't implement `forEach.` I'll convert to an array instead and use `forEach` for code consistency

Reply:
silly dom api